### DISCUSSION TITLE
Export Href type definition that works for getPathname, redirect & <Link /> params/props

### DISCUSSION THREAD

**Discussion by ferares** [amannn/next-intl#1280] [Q&A] [answered]
### Is your feature request related to a problem? Please describe.

I'm struggling to define an `Href` type that I can then on the return values of `createLocalizedPathnamesNavigation` (`getPathname`, `Link` and `redirect`)

I've defined it as `export type Href = Parameters<typeof getPathname>[0]["href"]` which seems to work fine for `getPathname` but it does not work for either `redirect` nor the `Link` component.

### Describe the solution you'd like

An importable Href type that's compatible with the `href` prop of the `Link` component and the `href` param of both `getPathname` and `redirect` functions.

### Describe alternatives you've considered

As of now I'm defining the Href type as `Parameters<typeof getPathname>[0]["href"]` and using `// @ts-expect-error` to silence errors when passing values into `Link` components or `redirect` calls

**Comment by amannn** [comment]
The `Href` type is not intended to be exported. You can likely define a better one based on what is shown in [the sitemap docs](https://next-intl-docs.vercel.app/docs/environments/actions-metadata-route-handlers#sitemap) for localized pathnames that considers the actual pathnames that are available in your app.

I'm going to move this to a discussion since this seems to be more of a usage question.

**Comment by ferares** [comment]
I have defined the `Href` type following the sitemap docs example but that type is not compatible with the prop on the `Link` component nor the `redirect` function's href param. Is there a way to define a type that would work on all 3  (`getPathname`, `Link` and `redirect`) and at the same time have the linter error out when using values that are not declared as valid paths?

**Reply by fnowacki** [reply]
yeah, it would be nice to define a type that would work on all places :+1:

@amannn could you help?

**Comment by ferares** [comment, marked as answer]
This seems to have been addressed in the latest version so I'm closing the discussion. Thanks! 

Refer to https://next-intl-docs.vercel.app/blog/next-intl-3-22 for more details
